src/runtime/arc.ts

```diff
@@ -134,7 +134,8 @@ export class Arc implements ArcInterface {
     this.inspector = inspectorFactory && inspectorFactory.create(this);
     this.storageKey = storageKey;
     const ports = this.pecFactories.map(f => f(this.generateID(), this.idGenerator));
-    this.pec = new ParticleExecutionHost(slotComposer, this, ports);
+    this.pec = new ParticleExecutionHost(
+        slotComposer, ports.map(port => new PECOuterPortImpl(port, this)));
```

Comment:
We break the encapsulation of ParticleExecutionHost by making PECOuterPortImpl visible... But I don't have a good suggestion how to improve things, largely because I don't understand the rules of cycle detection TS compiler in g3 follows.

Reply:
is the encapsulation here a concern? @shans 
in general, are we ok with merging this now, or do we want to continue investigating?

Reply:
We should add a TODO, this does seem like something we'd like to keep hidden encapsulation-wise so we should find another way but it doesn't seem like it has to block the cycle work.